x-pack/plugins/upgrade_assistant/__jest__/client_integration/overview/fix_issues_step/kibana_deprecation_issues.test.tsx

```diff
@@ -0,0 +1,132 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { act } from 'react-dom/test-utils';
+import { deprecationsServiceMock } from 'src/core/public/mocks';
+import type { DomainDeprecationDetails } from 'kibana/public';
+
+import { setupEnvironment, kibanaDeprecationsServiceHelpers } from '../../helpers';
+import { OverviewTestBed, setupOverviewPage } from '../overview.helpers';
+import { esNoDeprecations } from './mock_es_issues';
+
+describe('Overview - Fix deprecation issues step - Kibana deprecations', () => {
+  let testBed: OverviewTestBed;
+  const { server, httpRequestsMockHelpers } = setupEnvironment();
+  const {
+    mockedKibanaDeprecations,
+    mockedKibanaCriticalOnlyDeprecations,
+  } = kibanaDeprecationsServiceHelpers.defaultMockedResponses;
+
+  afterAll(() => {
+    server.restore();
+  });
+
+  describe('When load succeeds', () => {
+    const setup = async (response: DomainDeprecationDetails[]) => {
+      // Set up with no ES deprecations.
+      httpRequestsMockHelpers.setLoadEsDeprecationsResponse(esNoDeprecations);
+
+      await act(async () => {
+        const deprecationService = deprecationsServiceMock.createStartContract();
+        kibanaDeprecationsServiceHelpers.setLoadDeprecations({ deprecationService, response });
+
+        testBed = await setupOverviewPage({
+          services: {
+            core: {
+              deprecations: deprecationService,
+            },
+          },
+        });
+      });
+
+      const { component } = testBed;
+      component.update();
+    };
+
+    describe('when there are critical and warning issues', () => {
+      beforeEach(async () => {
+        await setup(mockedKibanaDeprecations);
+      });
+
+      test('renders counts for both', () => {
+        const { exists, find } = testBed;
+
+        expect(exists('kibanaStatsPanel')).toBe(true);
+        expect(find('kibanaStatsPanel.criticalDeprecations').text()).toContain(1);
+        expect(find('kibanaStatsPanel.warningDeprecations').text()).toContain(1);
```

Comment:
These assertions were originally against `mockedCriticalKibanaDeprecations.length` and `mockedWarningKibanaDeprecations.length`, but this test is intended to be about non-zero values and I think it's easier to understand that intention if we're explicit about the value we expect.